esp-hal-common/src/rsa/esp32.rs

```diff
@@ -139,6 +139,20 @@ where
         Ok(())
     }
 
+    #[cfg(feature = "async")]
+    pub async fn start_step2_async(&mut self, operand_b: &T::InputType) {
```

Comment:
```suggestion
    pub async fn start_step2(&mut self, operand_b: &T::InputType) {
```

Let's remove the async postfix.

Reply:
Infact, this whole function can be removed because its the same as `start_step2` (there are no await points).

Reply:
Just to have this documented:
We decided not to mix `async` and `nb` (which is required with the actual implementation) and make `start_step2` function a blocking function, but use the loop idea from my `start_step2_async` function